### DISCUSSION TITLE
UNION ALL of two SELECTs is not working within queryText

### DISCUSSION THREAD

**Discussion by ayoUbuntu** [eclipse-birt/birt#847] [Q&A]
I am trying to merge two select queries within the same dataset. I am only able to run one select query at a time but not their union.

Please have a look first at the file attached.
[report.txt](https://github.com/eclipse/birt/files/8187781/report.txt)

If I delete the beforeOpen and afterOpen scripts and delete the `UNION ALL buchungenQueryText` from the queryText then the report is generated and has got data in it. However when I use the template attached I get an emty report and no error messages. Any help will be more than appreciated.

**Comment by hvbtup** [comment]
This is probably a problem with your SQL and not with BIRT.

You didn't attach what your SQL text actually ist. Since you output it to System.out, you should consider attaching it to this issue - and test it outside of BIRT with a tool like SQL*Developer or whatever your DB offers.

Furthermore I see that you are passing input parameters as parts of a string concatenation. This is bad practice for two reasons:
1) SQL injection
2) Can be slower (as a rule of thumb: use bind variables, however, for top-level queries it sometimes make sense to pass the actual values to help the optimizer).
